### PR TITLE
[vcpkg] initial openbsd (community) support

### PR DESCRIPTION
**Pull request by jgilje**
**Describe the pull request**
This PR adds OpenBSD support for vcpkg. I've tested a couple of packages so far, which compile fine (libpq, botan, brotli, sdl2 with dependencies).The triplet-definition is added in the community folder. The patch consists mostly of a set of defines, I was helped in large part by looking at the previous FreeBSD support.

I've defaulted the port to use clang as the gcc included with OpenBSD is quite old (the last GPL2 release). In vcpkg_configure_make(), I've changed the behaviour when running configure to lookup bash in path first.

Quick instructions (included in the triplet file)
Bootstrap using
`CC=/usr/bin/clang CXX=/usr/bin/clang++ ./bootstrap-vcpkg.sh`
Use with
`VCPKG_FORCE_SYSTEM_BINARIES=1 ./vcpkg install brotli `

- What does your PR fix?
OpenBSD support for vcpkg

- Which triplets are supported/not supported? Have you updated the CI baseline?
x64-openbsd

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: scripts/cmake/vcpkg_configure_meson.cmake [hunk 1]
```diff
@@ -54,6 +54,8 @@ function(vcpkg_configure_meson)
             set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${SCRIPTS}/toolchains/osx.cmake")
         elseif(VVCPKG_TARGET_IS_FREEBSD)
             set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${SCRIPTS}/toolchains/freebsd.cmake")
+        elseif(VVCPKG_TARGET_IS_OPENBSD)
```
Comment (strega-nil) [inline scripts/cmake/vcpkg_configure_meson.cmake:57]: I think these should be `VCPKG_TARGET_...`

File: toolsrc/src/vcpkg/base/system.process.cpp [hunk 2]
```diff
@@ -156,6 +156,12 @@ namespace vcpkg
         Checks::check_exit(VCPKG_LINE_INFO, rcode == 0, "Could not determine current executable path.");
         Checks::check_exit(VCPKG_LINE_INFO, len > 0, "Could not determine current executable path.");
         return fs::path(exePath, exePath + len - 1);
+#elif defined(__OpenBSD__)
+        const char* progname = getprogname();
+        char resolved_path[PATH_MAX];
+        auto ret = realpath(progname, resolved_path);
+        Checks::check_exit(VCPKG_LINE_INFO, ret != nullptr, "Could not determine current executable path.");
+        return fs::path(std::string(resolved_path));
```
Comment (strega-nil) [inline toolsrc/src/vcpkg/base/system.process.cpp:164]:
```suggestion
        return fs::u8path(resolved_path);
```

File: scripts/toolchains/openbsd.cmake [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    string(APPEND CMAKE_CXX_FLAGS_RELEASE_INIT " ${VCPKG_CXX_FLAGS_RELEASE} ")
+
+    string(APPEND CMAKE_SHARED_LINKER_FLAGS_INIT " ${VCPKG_LINKER_FLAGS} ")
+    string(APPEND CMAKE_EXE_LINKER_FLAGS_INIT " ${VCPKG_LINKER_FLAGS} ")
+    string(APPEND CMAKE_SHARED_LINKER_FLAGS_DEBUG_INIT " ${VCPKG_LINKER_FLAGS_DEBUG} ")
+    string(APPEND CMAKE_EXE_LINKER_FLAGS_DEBUG_INIT " ${VCPKG_LINKER_FLAGS_DEBUG} ")
+    string(APPEND CMAKE_SHARED_LINKER_FLAGS_RELEASE_INIT " ${VCPKG_LINKER_FLAGS_RELEASE} ")
+    string(APPEND CMAKE_EXE_LINKER_FLAGS_RELEASE_INIT " ${VCPKG_LINKER_FLAGS_RELEASE} ")
+endif()
+endif()
```
Comment (strega-nil) [inline scripts/toolchains/openbsd.cmake:31]:
```suggestion

endif()
```

File: scripts/toolchains/openbsd.cmake [hunk 4]
```diff
@@ -0,0 +1,31 @@
+if(NOT _VCPKG_OPENBSD_TOOLCHAIN)
+set(_VCPKG_OPENBSD_TOOLCHAIN 1)
```
Comment (strega-nil) [inline scripts/toolchains/openbsd.cmake:2]:
```suggestion
if(NOT _VCPKG_OPENBSD_TOOLCHAIN)
set(_VCPKG_OPENBSD_TOOLCHAIN 1)

```

File: scripts/toolchains/openbsd.cmake [hunk 5]
```diff
@@ -0,0 +1,31 @@
+if(NOT _VCPKG_OPENBSD_TOOLCHAIN)
+set(_VCPKG_OPENBSD_TOOLCHAIN 1)
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "OPENBSD")
```
Comment (strega-nil) [inline scripts/toolchains/openbsd.cmake:3]: I think this should be `CMAKE_HOST_SYSTEM_NAME STREQUAL "OpenBSD"`, right?

File: scripts/toolchains/openbsd.cmake [hunk 6]
```diff
@@ -0,0 +1,31 @@
+if(NOT _VCPKG_OPENBSD_TOOLCHAIN)
+set(_VCPKG_OPENBSD_TOOLCHAIN 1)
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "OPENBSD")
+    set(CMAKE_CROSSCOMPILING OFF CACHE BOOL "")
+endif()
+set(CMAKE_SYSTEM_NAME OPENBSD CACHE STRING "")
```
Comment (strega-nil) [inline scripts/toolchains/openbsd.cmake:6]: Same here, shouldn't this be `"OpenBSD"`?

**Review by strega-nil — CHANGES_REQUESTED** [review]
This is great! Thanks so much for working on this, just some minor review comments 😄

**Comment by jgilje** [comment]
@strega-nil Thanks, I'll work through your review. I also forgot to add a patched pkg-config to the PR (They ship a custom pkg-config implementation which lacks the `{fcfiledir}` variable, I hope I'll be able to upstream these changes in the future).

**Comment by PhoebeHui** [comment]
Please ignore the CI failures, it's not relate to this change, I will rerun the PR after it fixed in baseline.

**Review by strega-nil — APPROVED** [review]
I'll run the scripts to update the docs before merging :)

**Comment by strega-nil** [comment]
Waiting on PR #14549

File: scripts/cmake/vcpkg_configure_meson.cmake [hunk 7]
```diff
@@ -143,6 +143,8 @@ function(vcpkg_internal_meson_generate_native_file_config _config) #https://meso
             set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${SCRIPTS}/toolchains/ios.cmake")
         elseif(VCPKG_TARGET_IS_FREEBSD)
             set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${SCRIPTS}/toolchains/freebsd.cmake")
+        elseif(VCPKG_TARGET_IS_OPENBSD)
+            set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${SCRIPTS}/toolchains/freebsd.cmake")
```
Comment (TorMFinn) [inline scripts/cmake/vcpkg_configure_meson.cmake:147]: Suppose the toolchain should be openbsd here and not freebsd?

Comment (strega-nil) [reply scripts/cmake/vcpkg_configure_meson.cmake:147]:
```suggestion
            set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE "${SCRIPTS}/toolchains/openbsd.cmake")
```

Comment (strega-nil) [reply scripts/cmake/vcpkg_configure_meson.cmake]: Thanks @TorMFinn

**Comment by strega-nil** [comment]
@jgilje thanks so much!

**Pull request merged**